include/mbgl/actor/scheduler.hpp

```diff
@@ -34,6 +39,24 @@ class Scheduler {
     virtual ~Scheduler() = default;
     virtual void schedule(std::weak_ptr<Mailbox>) = 0;
     
+    /*
+     A token to the scheduled action, at the end of
+     the token's life-time, the scheduled action is
+     canceled.
+    */
+    class Scheduled : private util::noncopyable {
+    public:
+        virtual ~Scheduled() = default;
+        
+        // Explicitly cancel the scheduled action
+        virtual void cancel() = 0;
+        
+        virtual bool isFinished() = 0;
+    };
```

Comment:
We already have an almost identical class called `WorkTask` (and a few others that are similar). Maybe we can reuse/unify those?